tic_tac_toe/game.py

```diff
@@ -87,18 +147,53 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    board = game["board"]
+    nextTurn = game["next_turn"]
+    playerOne = game["player1"]
+    playerTwo = game["player2"]
+    weHaveWinner = game["winner"]
+    
+    if (_board_is_full(board) or weHaveWinner):
+        raise InvalidMovement("Game is over.")
+    elif not (_position_is_valid(position)):
+        raise InvalidMovement("Position out of range.") 
+    elif not (_position_is_empty_in_board(position, board)):
+        raise InvalidMovement("Position already taken.")
+    elif (nextTurn != player):
+        raise InvalidMovement("\"{}\" moves next".format(nextTurn))
+    else:
+        board[position[0]][position[1]] = player
+        
+        playerWon = _check_winning_combinations(board, player)
+        if playerWon:
+            game["winner"] = player
+            raise GameOver("\"{}\" wins!".format(player))
+        if _board_is_full(board): 
+            raise GameOver("Game is tied!")
+        _swapTurn(game, player, playerOne, playerTwo)
 
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+    board = game["board"]
+
+    currentBoard = """
+{0}  |  {1}  |  {2}
+--------------
+{3}  |  {4}  |  {5}
+--------------
+{6}  |  {7}  |  {8}
+"""
+    currentBoard = currentBoard.format(board[0][0], board[0][1], board[0][2],
```

Comment:
Didn't know you could do this! I thought you needed to call `.format()` immediately after the string, but this makes sense and keeps the code cleaner and easier to read.